datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -3049,17 +3051,54 @@ pub enum Partitioning {
     DistributeBy(Vec<Expr>),
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum ColumnUnnestType {
+    List(Vec<ColumnUnnestList>),
+    // for struct, there can only be one unnest performed on one column at a time
+    Struct,
+    // Infer the unnest type based on column schema
+    // If column is a list column, the unnest depth will be 1
+    // This value is to support sugar syntax of old api (unnest(columns1,columns2))
+    Inferred,
+}
+
+impl fmt::Display for ColumnUnnestType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ColumnUnnestType::List(lists) => {
+                let list_strs: Vec<String> =
+                    lists.iter().map(|list| list.to_string()).collect();
+                write!(f, "List([{}])", list_strs.join(", "))
+            }
+            ColumnUnnestType::Struct => write!(f, "Struct"),
+            ColumnUnnestType::Inferred => write!(f, "Inferred"),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
```

Comment:
Can we please add documentation to this struct about what it means (what computation it represents)? In the context of this PR it is clear, but once we merge I think the context will be lost

Reply:
Thank you, comment addressed